src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,765 @@
+                }
+            }
+        }
+
+        return inputFileTypeMap;
+    }
+
+    /**
+     * Processes a single ACH input file.
+     */
+    // Portions of this method are based on the code and logic from CustomerLoadServiceImpl.loadFile
+    protected int loadACHBatchDetailFile(String inputFileName, BatchInputFileType batchInputFileType) {
+        int result = numFailedRows = 0;
+        
+        byte[] fileByteContent = safelyLoadFileBytes(inputFileName);
+        
+        LOG.info("Attempting to parse the file.");
+        
+        Object parsedObject = null;
+        try {
+            parsedObject = batchInputFileService.parse(batchInputFileType, fileByteContent);
+        } catch (ParseException e) {
+            String errorMessage = "Error parsing batch file: " + e.getMessage();
+            LOG.error(errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        }
+        
+        if (!(parsedObject instanceof List)) {
+            String errorMessage = "Parsed file was not of the expected type.  Expected [" + List.class + "] but got [" + parsedObject.getClass() + "].";
+            criticalError(errorMessage);
+        }
+        
+        @SuppressWarnings("unchecked")
+        List<PayeeACHAccountExtractDetail> achDetails = (List<PayeeACHAccountExtractDetail>) parsedObject;
+        
+        for (PayeeACHAccountExtractDetail achDetail : achDetails) {
+            if (!processACHBatchDetail(achDetail)) {
+                numFailedRows++;
+            }
+        }
+        
+        return numFailedRows;
+    }
+
+    /**
+     * Processes a single ACH batch detail, and routes a Payee ACH Account maintenance document accordingly.
+     */
+    protected boolean processACHBatchDetail(PayeeACHAccountExtractDetail achDetail) {
+        LOG.info("Starting processACHBatchDetail for: " +achDetail.getLogData());
+
+        Person payee = personService.getPersonByPrincipalName(achDetail.getNetID());
+
+        if (!validateACHBatchDetail(achDetail, payee)) {
+            return false;
+        }
+        
+        // Check for existing ACH accounts.
+        PayeeACHAccount entityAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.ENTITY, payee.getEntityId(), getDirectDepositTransactionType());
+        PayeeACHAccount employeeAccount = achService.getAchInformation(
+                PayeeIdTypeCodes.EMPLOYEE, payee.getEmployeeId(), getDirectDepositTransactionType());
+        
+        String processingResults = KFSConstants.EMPTY_STRING;
+        // Add or update Entity ID account.
+        if (ObjectUtils.isNull(entityAccount)) {
+            processingResults = addACHAccount(payee, achDetail, PayeeIdTypeCodes.ENTITY);
+            LOG.info("Created new ACH Account of Entity type for payee " + payee.getPrincipalName());
+        } else {
+            processingResults = updateACHAccountIfNecessary(payee, achDetail, entityAccount);

```

Comment:
I think it would be cleaner to log the info about the processing in the add and update ACHAccount methods instead of returning a String to process and log here. 
I would change the add and update methods to either return a boolean or thow and exception that you catch here and log accordingly. Probably returning a boolean and logging the info about the processed data in the add/update methods would be better.